### ISSUE TITLE
unused-argument warnings are not reported if routine has "raise"

### ISSUE THREAD

**Issue by slawek-es** [PyCQA/pylint#3416]
It looks as though the presence of a `raise` statement makes Pylint omit some warnings, such as `unused-argument`. This seems to be a regression in `2.4.2` because this works in `2.4.1`.

Note that in the examples below, removing the `raise` makes Pylint complain about unused arguments again.

### Steps to reproduce
1. Try to lint the following code:

```
def fun(arg_a, arg_b, arg_c) -> None:
    """Routine docstring"""
    try:
        pass
    except Exception:
        raise RuntimeError("")
```

### Current behavior

```
$ python3 -m pylint /tmp/test.py 

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

The unused-argument warnings should be reported. This is coming from pylint `2.4.1`:
```
$ python3 -m pylint /tmp/test.py 
************* Module test
/tmp/test.py:3:8: W0613: Unused argument 'arg_a' (unused-argument)
/tmp/test.py:3:15: W0613: Unused argument 'arg_b' (unused-argument)
/tmp/test.py:3:22: W0613: Unused argument 'arg_c' (unused-argument)

-------------------------------------------------------------------
Your code has been rated at 4.00/10 (previous run: 10.00/10, -6.00)
```

### pylint --version output

```
$ pylint --version
pylint 2.4.2
astroid 2.3.3
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0]
```
